node/src/components/storage.rs

```diff
@@ -444,6 +451,11 @@ impl Storage {
         initialize_block_metadata_db(&env, &block_metadata_db, &deleted_block_hashes_raw)?;
         initialize_deploy_metadata_db(&env, &deploy_metadata_db, &deleted_deploy_hashes)?;
 
+        let mut metrics: Option<Metrics> = None;
+        if let Some(metrics_registry) = registry {
+            metrics = Some(Metrics::new(metrics_registry)?);
+        }
```

Comment:
Just to avoid the `mut`, you could maybe do this instead:
```suggestion
        let metrics = registry.map(Metrics::new).transpose()?;
```